### PR TITLE
hdf5 1.13.0

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Someone will have to fix the GDAL patch, or we need to wait for https://github.com/OSGeo/gdal/milestone/35

File: Formula/gdal.rb [hunk 1]
```diff
@@ -67,6 +67,12 @@ class Gdal < Formula
 
   fails_with gcc: "5"
 
+  # Support hdf5 1.13, remove in next release
+  patch do
```
[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/gdal.rb]: Thanks 👍

**Comment by cho-m** [comment]
Note that this PR will also have merge conflict with `icu4c` #88210 due to `vtk`.

**Review by Bo98 — DISMISSED** [review]

**Review by iMichka — DISMISSED** [review]

**Comment by iMichka** [comment]
`f3d` got rev-bumped for `opencascade`. This will need a rebase and a new rev-bump for `f3d`

**Review by carlocab — APPROVED** [review]

**Comment by samford** [comment]
@SMillerDev 1.13.0 is currently listed as "experimental" on the [first-party download page](https://www.hdfgroup.org/downloads/hdf5) and 1.12.1 is still given as the latest (stable?) version. Does the `hdf5` formula track experimental releases or was this PR an oversight?

If 1.13.0 is correct, I'll need to update the `livecheck` block to also match experimental releases.

**Comment by SMillerDev** [comment]
I'm pretty sure I used the livecheck version at the time. Strange that it's marked as experimental.

**Comment by Bo98** [comment]
This is the first ever expereimental release so it's possible it was initially deployed wrong.

Looks like they are going the old GNOME route: odd number minor versions are experimental and don't even guarantee ABI stability between patch releases.

**Comment by samford** [comment]
> I'm pretty sure I used the livecheck version at the time. Strange that it's marked as experimental.

> This is the first ever experimental release so it's possible it was initially deployed wrong.

1.13.0 was tagged on 2021-12-03 but a [2021-12-17 snapshot of the source code download page](https://web.archive.org/web/20211217144319/https://www.hdfgroup.org/downloads/hdf5/source-code/) (what livecheck checks) has the same content as the current page, where 1.12.1 is the newest release. There aren't any snapshots after 2021-12-17 but snapshots of the main download page show that it has listed 1.13.0 as experimental since at least 2021-12-09 onward. Anything's possible but it seems unlikely that the source code page would have been modified to include 1.13.0 between 2021-12-17 and when this PR was created on 2022-02-01.

If you use `brew bump`, it was likely reporting 1.13.0 as the latest Repology version when this PR was created (i.e., another repository updated to 1.13.0 in late January) and it's given as a stable version (not "devel"). I [reported the issue to Repology](https://repology.org/project/hdf5/report), so hopefully they'll sort that out. That said, the livecheck version was almost certainly still 1.12.1 at the time and `brew bump` would have listed that before the Repology version.

The good news is that the `livecheck` block is probably fine (though we could update the regex to only match even-numbered minor versions, to be on the safe side). The bad news is that we have to figure out how to handle this situation, as we're one of only three package managers using 1.13.0 (one of which is seemingly tracking unstable versions). For what it's worth, 1.12.1 was tagged on 2021-07-06 and 1.12.0 was tagged on 2020-03-02, so it may (or may not) be a while before a new stable version appears.

> Looks like they are going the old GNOME route: odd number minor versions are experimental and don't even guarantee ABI stability between patch releases.

The [full Downloads page for prior versions](https://portal.hdfgroup.org/display/support/Downloads) (both experimental and stable) seems to support this idea, as the listed versions before 1.13 are 1.8, 1.10, and 1.12.

**Comment by Bo98** [comment]
> he bad news is that we have to figure out how to handle this situation, as we're one of only three package managers using 1.13.0 (one of which is seemingly tracking unstable versions).

We should probably downgrade. 1.14 is a while away: late 2022.

**Comment by SMillerDev** [comment]
For nginx and gnutls we also follow their unstable release. If we don't get any complaints and everything builds fine let's just go for the latest version.

**Comment by Bo98** [comment]
Problem to me is ABI stability. Every patch update will require revision bumps, though it's not clear if they'll bump ABI versions or not.

https://www.hdfgroup.org/2021/12/hdf5-1-13-0-introducing-experimental-releases/

> binary compatibility should not be expected in an experimental release series (e.g., HDF5 1.13.1 is unlikely to be binary compatible with HDF5 1.13.0).

> users may also encounter forward/backward file format incompatibility when using new features.

**Comment by SMillerDev** [comment]
Yeah, that is going to be messy. I'd be opposed to downgrading for the sake of the name of the release, but this probably warrants a downgrade.

**Pull request closed without merging**